src/lib/structures/base/Store.ts

```diff
@@ -0,0 +1,188 @@
+			const LoadedPiece = 'default' in loaded ? loaded.default : loaded;
+			if (!isClass(LoadedPiece)) throw new TypeError('The exported structure is not a class.');
+			piece = this.set(new LoadedPiece(this, file, directory));
+		} catch (error) {
+			this.client.emit('wtf', `Failed to load file '${loc}'. Error:\n${error.stack || error}`);
+		}
+		delete require.cache[loc];
+		module.children.pop();
+		return piece;
+	}
+
+	/**
+	 * Loads all of our Pieces from both the user and core directories.
+	 * @since 0.0.1
+	 * @returns The number of Pieces loaded.
+	 */
+	public async loadAll(): Promise<number> {
+		this.clear();
+		if (!this.client.options.disabledCorePieces.includes(this.name)) {
+			for (const directory of this.coreDirectories) await Store.walk(this, directory);
+		}
+		await Store.walk(this);
+		return this.size;
+	}
+
+	/**
+	 * Sets up a piece in our store.
+	 * @since 0.0.1
+	 * @param piece The piece we are setting up
+	 */
+	// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+	// @ts-ignore
+	public set(piece: V): V | null {
+		if (!(piece instanceof this.holds)) {
+			this.client.emit('error', `Only ${this} may be stored in this Store.`);
+			return null;
+		}
+
+		const existing = this.get(piece.name);
+		if (existing) this.delete(existing);
+
+		this.client.emit('pieceLoaded', piece);
+		super.set(piece.name, piece);
+		return piece;
+	}
+
+	/**
+	 * Deletes a command from the store.
+	 * @since 0.0.1
+	 * @param name A command object or a string representing a command or alias name
+	 * @returns Whether or not the delete was successful.
+	 */
+	public delete(name: V | string): boolean {
+		const piece = this.resolve(name);
+		if (!piece) return false;
+		super.delete(piece.name);
+		return true;
+	}
+
+	/**
+	 * Resolve a string or piece into a piece object.
+	 * @since 0.0.1
+	 * @param name The piece object or a string representing a piece's name
+	 */
+	public resolve(name: V | string): Piece | null {
+		if (name instanceof this.holds) return name;
+		return this.get(name as string) || null;
+	}
+
+	/**
+	 * Defines toString behavior for stores
+	 * @since 0.0.1
+	 * @returns This store name
+	 */
+	public toString(): string {
+		return this.name;
+	}
+
+	/**
+	 * Walks our directory of Pieces for the user and core directories.
+	 * @since 0.0.1
+	 * @param store The store we're loading into
+	 * @param directory The directory to walk in
+	 */
+	private static async walk<T extends Piece>(store: Store<T>, directory: string = store.userDirectory): Promise<T[]> {
+		const files = await scan(directory, { filter: (stats, path) => stats.isFile() && extname(path) === '.js' })
+			.catch(() => { if (store.client.options.createPiecesFolders) ensureDir(directory).catch(err => store.client.emit('error', err)); });
+		if (!files) return [];
+
+		return Promise.all([...files.keys()].map(file => store.load(directory, relative(directory, file).split(sep)) as Promise<T>));
+	}
```

Comment:
I really think you should make these 2 lines (first and last in this block) not squashed into 1 very hard to read line, when it could be spread across several and be much easier to read (id also apply this review to any line of similar length)

Reply:
Resolved in 24978d2.